llama_stack/providers/inline/agents/meta_reference/persistence.py

```diff
@@ -183,21 +183,39 @@ async def get_num_infer_iters_in_turn(self, session_id: str, turn_id: str) -> in
         )
         return int(value) if value else None
 
-    async def list_sessions(self) -> list[Session]:
-        values = await self.kvstore.values_in_range(
-            start_key=f"session:{self.agent_id}:",
-            end_key=f"session:{self.agent_id}:\xff\xff\xff\xff",
-        )
+    async def _get_sessions_from_keys(self, keys: list[str]) -> list[Session]:
```

Comment:
unrelated and definitely backwards-incompat but I really think KVStore was a bad choice here. Too much denormalization. Now that we have sqlstore, we should use that I think. Thoughts? (also cc @ehhuang) We'd probably need a migration etc.

Reply:
Shall we file an issue for this?

Reply:
Created one here:  https://github.com/meta-llama/llama-stack/issues/3130

Reply:
Will tag some folks.